### PR TITLE
Improvements to the ready 'rdy' commandline utility.

### PR DESCRIPTION
**Pull request by danwills**
In this revision, Ready's commandline utility 'rdy', includes boost as a link dependency, and can do much more than it did before.

I will soon also be able to contribute several Houdini Digital Assets (HDA's) that will allow you (by pointing at this rdy binary) to import OpenCLImageRD-based formulas directly into Houdini as Gas OpenCL Dops.

This is what the help (-h or --help argument) says in this version:
```
========================
Ready commandline utility that uses readybase as a processing (and data-retrieval) back-end.

It responds to various commandline arguments to load and print out parts of the system,
and can also still be used (like it used to work) to iterate the system to a specified number
of steps, and save it to a new vti file.

The various print options can facilitate the import of ready simulations into other applications, (such
as Houdini), without the need to actually link the ready libraries. This includes reagent initial-states,
(via -m), so be ready for some large lumps of text on stdout when using that argument.

Do let the Ready team know if there is something that you wish to print that currently isn't supported.

Options:
  -h [ --help ]                        Print the help message
  -n [ --num-iterations ] arg (=0)     Number of iterations to run before
                                       saving
  -k [ --print-kernel ]                Print (full) OpenCL kernel source (when
                                       possible)
  -f [ --print-formula ]               Print kernel formula (when possible)
  -r [ --print-reagent-info ]          Print reagent info
  -p [ --print-parameter-info ]        Print parameter info
  -s [ --print-render-settings ]       Print render Settings
  -d [ --print-formula-description ]   Print formula Description
  -m [ --print-initial-state-images ]  Print initial state images (Warning: May
                                       be large!)
  -i [ --vti-in ] arg                  VTI file to load (required)
  -o [ --vti-out ] arg                 VTI file to save (optional)
  -l [ --opencl-platform ] arg         OpenCL platform number (Currently will
                                       crash if incorrect!)
  -g [ --opencl-device ] arg           OpenCL device number (Currently will
                                       crash if incorrect!)
  -v [ --verbose ]                     Verbose output.
========================
```

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -479,9 +479,14 @@ else()
   )
  endif()
 endif()
+
 # create command-line utility
 add_executable( ${CMD_NAME} ${CMD_SOURCES} )
-target_link_libraries( ${CMD_NAME} readybase )
+target_link_libraries( ${CMD_NAME} readybase boost_program_options)
+
+# Link boost >1.65 (the improvements to the 'rdy' binary will not build without it).
```
Comment (timhutton) [inline CMakeLists.txt]: Can this package requirement be relaxed to an older version? Our Travis Continuous Integration uses Ubuntu 14.04 (trusty), which comes with Boost 1.54. Travis have very recently (Nov 8) added Ubuntu 16.04 (xenial) which comes with Boost 1.58.

Comment (danwills) [reply CMakeLists.txt]: Thanks for taking a look Tim. I had previously tried to figure out the minimum version we would need and 1.65 is what I ended up with but now I think I was wrong. I've read one page now that seemed to be suggesting that 1.50 would be sufficient. I have committed and pushed a downgrade of the requirement to that. Do I need to redo the pull request?

Comment (timhutton) [reply CMakeLists.txt]: No need to redo the PR. Github picks up changes to the branch when pushed and updates the PR for you. It also triggers a fresh run of Travis.

File: CMakeLists.txt [hunk 2]
```diff
@@ -454,7 +454,7 @@ else()
     vtkRenderingCore
     vtkRenderingAnnotation
     vtkRenderingFreeType
-    vtkRenderingOpenGL
+    vtkRenderingOpenGL2
```
Comment (timhutton) [inline CMakeLists.txt]: Do you need to change this? This section only applies if you are using a version of VTK that is between 6.3 and 7. What is your platform and VTK version?

File: CMakeLists.txt [hunk 3]
```diff
@@ -413,9 +413,9 @@ if( EXISTS "/etc/redhat-release" )
  # This *should* work everywhere but causes problems on Debian 8 and Travis-CI
  target_link_libraries( readybase ${VTK_LIBRARIES} )
 else()
- if( UNIX AND NOT APPLE )
-  set( EXTRA_VTK_LIBS vtkIOMPIImage vtkIOMPIParallel vtkRenderingFreeTypeFontConfig vtkRenderingMatplotlib )
- endif()
+ #if( UNIX AND NOT APPLE )
+ # set( EXTRA_VTK_LIBS vtkIOMPIImage vtkIOMPIParallel vtkRenderingFreeTypeFontConfig vtkRenderingMatplotlib )
+ #endif()
  if( ${VTK_MAJOR_VERSION} LESS 6 )
```
Comment (timhutton) [inline CMakeLists.txt]: The commit that added these lines was this: https://github.com/GollyGang/ready/commit/0eeb9966a9b5e6d12fa3d8b8a556a2c1c15f32a6
(git/github has a 'blame' feature that is useful for finding these)

What platform and VTK version did these lines cause a problem with?

File: CMakeLists.txt (hunk 2 above)
Comment (timhutton) [reply CMakeLists.txt]: Also this change breaks the build on Ubuntu 17.10 for me (VTK 6.3).

**Comment by timhutton** [comment]
Can you add one line in Help/changes.html to say what changed?

Can you add some instructions somewhere in Ready's help system to explain how to use rdy with Houdini?

File: CMakeLists.txt (hunk 2 above)
Comment (danwills) [reply CMakeLists.txt]: Yep totally didn't need to happen, must have picked it up along the way.
Apologies!

**Review by danwills — COMMENTED** [review]
Thanks for all the help with landing this Tim! I will be switching to an editor with better whitespace soon.

**Comment by danwills** [comment]
Woo all checks passed!
Thanks again for all your help @timhutton ! champion!

**Pull request merged**